src/DynamoWebServer/Messages/MessageHandler.cs

```diff
@@ -465,6 +473,44 @@ private string GetValue(NodeModel node)
             return data;
         }
 
+        private object GenerateWatchData(MirrorData data)
+        {
+            if (data.IsCollection)
+            {
+                var list = new List<object>();
+                var elements = data.GetElements();
+
+                foreach (var mirrorData in elements)
+                {
+                    list.Add( mirrorData.IsCollection
+                            ? GenerateWatchData(mirrorData)
+                            : GetSingleObjectPreview(mirrorData));
+                }
+
+                return list.ToArray();
+            }
+
+            return GetSingleObjectPreview(data);
+        }
+
+        private object GetSingleObjectPreview(MirrorData data)
+        {
+            if (data.Data != null)
+            {
+                double number;
+                return double.TryParse(data.StringData, out number)
+                    ? data.Data
+                    : data.Data.ToString();
```

Comment:
If we return data.StringData in Json it will be sent like "\"42\"", but when we return data.Data it will be sent like "42" and in Flood it will be parsed as number and we'll be able to apply special formatting for it view.